packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -0,0 +1,118 @@
+import { createRule } from '../util';
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+
+export default createRule({
```

Comment:
You should annotate the types specifically here as TS will otherwise infer the type of the options to be `unknown[]`.

```suggestion

type MessageIds = 'preferRecord' | 'preferIndexSignature';
type Options = [
  'always-record' | 'always-index-sig',
];

export default createRule<Options, MessageIds>({
```